website/docs: "Device code flow" documentation improvements (#14257)
goauthentik/authentik: website/docs/add-secure-apps/providers/oauth2/device_code.md

# Device code flow

(Also known as device flow, Device Authorization Grant flow, and [RFC 8628](https://datatracker.ietf.org/doc/html/rfc8628))

> This type of authentication flow is useful for devices with limited input capabilities and/or devices without browsers.

Reviewer: ```suggestion
```

> The OAuth 2.0 device authorization grant is designed for Internet-connected
> devices that either lack a browser to perform a user-agent-based
> authorization or are input constrained to the extent that
> requiring the user to input text in order to authenticate during the
> authorization flow is impractical. It enables OAuth clients on such
> devices (like smart TVs, media consoles, digital picture frames, and
> printers) to obtain user authorization to access protected resources
> by using a user agent on a separate device.

### Requirements

> This device flow is only possible if the active [brand](../../../sys-mgmt/brands.md) has a device code flow setup. This flow is run _after_ the user logs in, and before the user authenticates.

Reviewer: ```suggestion
This device flow is only possible if the active [brand](../../../sys-mgmt/brands.md) has a device code flow configured. This flow is run _after_ the user logs in, and before the user authenticates.
```

Reviewer: ...or "set up" if you prefer.

> authentik does not include a default flow for this use case, so it is necessary to create a new one with the **Designation** of **Stage Configuration**.

Reviewer: ```suggestion
authentik does not include a default flow for this use case, so it is necessary to create a new one with the **Designation** of `Stage Configuration`.
```

# Device code flow

> (Also known as device flow, Device Authorization Grant flow and [RFC 8628](https://datatracker.ietf.org/doc/html/rfc8628))

Reviewer 2: ```suggestion
(Also known as device flow, Device Authorization Grant flow, and [RFC 8628](https://datatracker.ietf.org/doc/html/rfc8628))
```

This type of authentication flow is useful for devices with limited input capabilities and/or devices without browsers.

> > The OAuth 2.0 device authorization grant is designed for Internet-connected

Reviewer 2: I think it's best to link the source of this quote block rather than including it here

Author: @dominic-r I have fully quoted it for indexing purposes, also, it's great to have a copy of the summary in the website imo

Reviewer: Depends.. if this block is truly a quote, from another source, then we can leave it as is (I do think it is good to share that info right here), but then we have to link to the source.

A better approach is to paraphrase it (and then we do not need to link, which is best to avoid for maintenance reasons). And I would remove the block quote formatting. Just provide the info.

Author: The quote is literally the abstract of https://datatracker.ietf.org/doc/html/rfc8628, that is exactly what we mention in the "also known as" part. I believe source has already been quoted at that point

Reviewer: Ah, now I see, thanks for the explanation. As a new reader, there was no way to associate the linked RFC source to the quote several lines down. Let's rearrange this section to make it more clear here that quote comes from. ;-)

+++++++++++++++++++

The device code flow is also known as _device flow_ or _device authorization grant flow_. This type of authentication flow is useful for devices with limited input capabilities and/or devices without browsers. The [Request for Comments (RFC) 8628](https://datatracker.ietf.org/doc/html/rfc8628)) abstract for this flow states:

> The OAuth 2.0 device authorization grant is designed for Internet-connected devices that either lack a browser to perform a user-agent-based authorization or are input constrained to the extent that requiring the user to input text in order to authenticate during the authorization flow is impractical. It enables OAuth clients on such devices (like smart TVs, media consoles, digital picture frames, and printers) to obtain user authorization to access protected resources by using a user agent on a separate device.

> devices that either lack a browser to perform a user-agent-based
> authorization or are input constrained to the extent that
> requiring the user to input text in order to authenticate during the
> authorization flow is impractical. It enables OAuth clients on such
> devices (like smart TVs, media consoles, digital picture frames, and
> printers) to obtain user authorization to access protected resources
> by using a user agent on a separate device.

### Requirements

This device flow is only possible if the active [brand](../../../sys-mgmt/brands.md) has a device code flow setup. This flow is run _after_ the user logs in, and before the user authenticates.

> authentik doesn't ship with a default flow for this usecase, so it is required to create a new flow for this usecase with the **Designation** of _Stage Configuration_.

Reviewer 2: ```suggestion
authentik does not include a default flow for this use case, so it is necessary to create a new one with the **Designation** of **Stage Configuration**.
```

### Steps

> 1. Go to **Admin Interface** > **Flows and Stages** > **Flows**

Reviewer 2: ```suggestion
1. From the authentik Admin interface, navigate to **Flows and Stages** > **Flows**.
```

> 2. Create new Flow, fill the form ensuring that **Designation** is set to _Stage Configuraton_

Reviewer 2: ```suggestion
2. Create a new Flow by clicking **Create**, and fill out the form ensuring **Designation** is set to `Stage Configuration`.
```

> 3. In **System** > **Brands** select the edit icon in **Actions** column

Reviewer 2: ```suggestion
3. Under **System** > **Brands**, select your brand and click the edit icon in **Actions** column.
```

> 4. Open **Default flows** section (from the accordion), select the newly created flow (from step 2.) in **Device code flow**.

Reviewer 2: ```suggestion
4. Under **Default flows**, select, under **Device code flow** the flow created in step 2.
```
